\hypertarget{classoomph_1_1OomphLibException}{}\doxysection{oomph\+::Oomph\+Lib\+Exception Class Reference}
\label{classoomph_1_1OomphLibException}\index{oomph::OomphLibException@{oomph::OomphLibException}}


//////////////////////////////////////////////////////////////////// ////////////////////////////////////////////////////////////////////  




{\ttfamily \#include $<$oomph\+\_\+definitions.\+h$>$}

Inheritance diagram for oomph\+::Oomph\+Lib\+Exception\+:\begin{figure}[H]
\begin{center}
\leavevmode
\includegraphics[height=2.991453cm]{classoomph_1_1OomphLibException}
\end{center}
\end{figure}
\doxysubsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
void \mbox{\hyperlink{classoomph_1_1OomphLibException_a92da7bc6b484044587ca2e8db4e1e507}{disable\+\_\+error\+\_\+message}} ()
\begin{DoxyCompactList}\small\item\em Suppress issueing of the error message in destructor (useful if error is caught successfully!) \end{DoxyCompactList}\end{DoxyCompactItemize}
\doxysubsection*{Protected Member Functions}
\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{classoomph_1_1OomphLibException_a2bcc47316fcc96608e79bec5ab49f82b}{Oomph\+Lib\+Exception}} (const std\+::string \&error\+\_\+description, const std\+::string \&function\+\_\+name, const char $\ast$location, const std\+::string \&exception\+\_\+type, std\+::ostream \&exception\+\_\+stream, const unsigned \&output\+\_\+width, bool list\+\_\+trace\+\_\+back)
\begin{DoxyCompactList}\small\item\em Constructor takes the error description, function name and a location string provided by the OOMPH\+\_\+\+EXCEPTION\+\_\+\+LOCATION macro and combines them into a standard header. The exception type will be the string \char`\"{}\+WARNING\char`\"{} or \char`\"{}\+ERROR\char`\"{} and the message is written to the exception\+\_\+stream, with a specified output\+\_\+width. Optionally provide a traceback of the function calls. \end{DoxyCompactList}\item 
\mbox{\hyperlink{classoomph_1_1OomphLibException_a0cc216d83aa854e965c3c8376d2c4d39}{$\sim$\+Oomph\+Lib\+Exception}} ()  throw ()
\begin{DoxyCompactList}\small\item\em The destructor cannot throw an exception (C++ STL standard) \end{DoxyCompactList}\end{DoxyCompactItemize}
\doxysubsection*{Protected Attributes}
\begin{DoxyCompactItemize}
\item 
std\+::ostream $\ast$ \mbox{\hyperlink{classoomph_1_1OomphLibException_ab9177b3f4d969f3e4d102487165103c7}{Exception\+\_\+stream\+\_\+pt}}
\begin{DoxyCompactList}\small\item\em Exception stream to which we write message in destructor. \end{DoxyCompactList}\item 
std\+::stringstream $\ast$ \mbox{\hyperlink{classoomph_1_1OomphLibException_a94df333a8bdcb4b89eeb516a64a3429e}{Exception\+\_\+stringstream\+\_\+pt}}
\begin{DoxyCompactList}\small\item\em String stream that records the error message. \end{DoxyCompactList}\item 
bool \mbox{\hyperlink{classoomph_1_1OomphLibException_ac374cd3358bf26639860a6778779786d}{Suppress\+\_\+error\+\_\+message}}
\begin{DoxyCompactList}\small\item\em Boolean to suppress issuing of the error message in destructor (useful if error is caught successfully!) \end{DoxyCompactList}\end{DoxyCompactItemize}


\doxysubsection{Detailed Description}
//////////////////////////////////////////////////////////////////// //////////////////////////////////////////////////////////////////// 

===================================================================== A Base class for oomph-\/lib run-\/time exception (error and warning) handling.

The class can only be instantiated by the derived classes \mbox{\hyperlink{classoomph_1_1OomphLibError}{Oomph\+Lib\+Error}} and \mbox{\hyperlink{classoomph_1_1OomphLibWarning}{Oomph\+Lib\+Warning}}. The (protected) constructor combines its string arguments into a standard format for uniform exception reports which are written to the specified output stream. 

Definition at line 173 of file oomph\+\_\+definitions.\+h.



\doxysubsection{Constructor \& Destructor Documentation}
\mbox{\Hypertarget{classoomph_1_1OomphLibException_a2bcc47316fcc96608e79bec5ab49f82b}\label{classoomph_1_1OomphLibException_a2bcc47316fcc96608e79bec5ab49f82b}} 
\index{oomph::OomphLibException@{oomph::OomphLibException}!OomphLibException@{OomphLibException}}
\index{OomphLibException@{OomphLibException}!oomph::OomphLibException@{oomph::OomphLibException}}
\doxysubsubsection{\texorpdfstring{OomphLibException()}{OomphLibException()}}
{\footnotesize\ttfamily oomph\+::\+Oomph\+Lib\+Exception\+::\+Oomph\+Lib\+Exception (\begin{DoxyParamCaption}\item[{const std\+::string \&}]{error\+\_\+description,  }\item[{const std\+::string \&}]{function\+\_\+name,  }\item[{const char $\ast$}]{location,  }\item[{const std\+::string \&}]{exception\+\_\+type,  }\item[{std\+::ostream \&}]{exception\+\_\+stream,  }\item[{const unsigned \&}]{output\+\_\+width,  }\item[{bool}]{list\+\_\+trace\+\_\+back }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [protected]}}



Constructor takes the error description, function name and a location string provided by the OOMPH\+\_\+\+EXCEPTION\+\_\+\+LOCATION macro and combines them into a standard header. The exception type will be the string \char`\"{}\+WARNING\char`\"{} or \char`\"{}\+ERROR\char`\"{} and the message is written to the exception\+\_\+stream, with a specified output\+\_\+width. Optionally provide a traceback of the function calls. 

The \mbox{\hyperlink{classoomph_1_1OomphLibException}{Oomph\+Lib\+Exception}} constructor takes the error description, function name, a location string provided by the OOMPH\+\_\+\+EXCEPTION\+\_\+\+LOCATION and an exception type \char`\"{}\+WARNING\char`\"{} or \char`\"{}\+ERROR\char`\"{} and combines them into a standard error message that is written to the exception stream. The output\+\_\+width of the message can also be specified. Optionally provide a traceback of the function calls. 

Definition at line 144 of file oomph\+\_\+definitions.\+cc.



References Exception\+\_\+stream\+\_\+pt, Exception\+\_\+stringstream\+\_\+pt, i, print\+\_\+stacktrace(), oomph\+::\+Global\+\_\+string\+\_\+for\+\_\+annotation\+::string(), and Suppress\+\_\+error\+\_\+message.

\mbox{\Hypertarget{classoomph_1_1OomphLibException_a0cc216d83aa854e965c3c8376d2c4d39}\label{classoomph_1_1OomphLibException_a0cc216d83aa854e965c3c8376d2c4d39}} 
\index{oomph::OomphLibException@{oomph::OomphLibException}!````~OomphLibException@{$\sim$OomphLibException}}
\index{````~OomphLibException@{$\sim$OomphLibException}!oomph::OomphLibException@{oomph::OomphLibException}}
\doxysubsubsection{\texorpdfstring{$\sim$OomphLibException()}{~OomphLibException()}}
{\footnotesize\ttfamily oomph\+::\+Oomph\+Lib\+Exception\+::$\sim$\+Oomph\+Lib\+Exception (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption}) throw ( ) \hspace{0.3cm}{\ttfamily [protected]}}



The destructor cannot throw an exception (C++ STL standard) 

//////////////////////////////////////////////////////////////////// ////////////////////////////////////////////////////////////////////

The \mbox{\hyperlink{classoomph_1_1OomphLibException}{Oomph\+Lib\+Exception}} destructor actually spawns the error message created in the constructor (unless suppresed) 

Definition at line 126 of file oomph\+\_\+definitions.\+cc.



References Exception\+\_\+stringstream\+\_\+pt, and Suppress\+\_\+error\+\_\+message.



\doxysubsection{Member Function Documentation}
\mbox{\Hypertarget{classoomph_1_1OomphLibException_a92da7bc6b484044587ca2e8db4e1e507}\label{classoomph_1_1OomphLibException_a92da7bc6b484044587ca2e8db4e1e507}} 
\index{oomph::OomphLibException@{oomph::OomphLibException}!disable\_error\_message@{disable\_error\_message}}
\index{disable\_error\_message@{disable\_error\_message}!oomph::OomphLibException@{oomph::OomphLibException}}
\doxysubsubsection{\texorpdfstring{disable\_error\_message()}{disable\_error\_message()}}
{\footnotesize\ttfamily void oomph\+::\+Oomph\+Lib\+Exception\+::disable\+\_\+error\+\_\+message (\begin{DoxyParamCaption}{ }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}



Suppress issueing of the error message in destructor (useful if error is caught successfully!) 



Definition at line 178 of file oomph\+\_\+definitions.\+h.



References Suppress\+\_\+error\+\_\+message, and oomph\+::\+Terminate\+Helper\+::suppress\+\_\+exception\+\_\+error\+\_\+messages().



Referenced by oomph\+::\+Finite\+Element\+::locate\+\_\+zeta().



\doxysubsection{Member Data Documentation}
\mbox{\Hypertarget{classoomph_1_1OomphLibException_ab9177b3f4d969f3e4d102487165103c7}\label{classoomph_1_1OomphLibException_ab9177b3f4d969f3e4d102487165103c7}} 
\index{oomph::OomphLibException@{oomph::OomphLibException}!Exception\_stream\_pt@{Exception\_stream\_pt}}
\index{Exception\_stream\_pt@{Exception\_stream\_pt}!oomph::OomphLibException@{oomph::OomphLibException}}
\doxysubsubsection{\texorpdfstring{Exception\_stream\_pt}{Exception\_stream\_pt}}
{\footnotesize\ttfamily std\+::ostream$\ast$ oomph\+::\+Oomph\+Lib\+Exception\+::\+Exception\+\_\+stream\+\_\+pt\hspace{0.3cm}{\ttfamily [protected]}}



Exception stream to which we write message in destructor. 



Definition at line 206 of file oomph\+\_\+definitions.\+h.



Referenced by Oomph\+Lib\+Exception().

\mbox{\Hypertarget{classoomph_1_1OomphLibException_a94df333a8bdcb4b89eeb516a64a3429e}\label{classoomph_1_1OomphLibException_a94df333a8bdcb4b89eeb516a64a3429e}} 
\index{oomph::OomphLibException@{oomph::OomphLibException}!Exception\_stringstream\_pt@{Exception\_stringstream\_pt}}
\index{Exception\_stringstream\_pt@{Exception\_stringstream\_pt}!oomph::OomphLibException@{oomph::OomphLibException}}
\doxysubsubsection{\texorpdfstring{Exception\_stringstream\_pt}{Exception\_stringstream\_pt}}
{\footnotesize\ttfamily std\+::stringstream$\ast$ oomph\+::\+Oomph\+Lib\+Exception\+::\+Exception\+\_\+stringstream\+\_\+pt\hspace{0.3cm}{\ttfamily [protected]}}



String stream that records the error message. 



Definition at line 209 of file oomph\+\_\+definitions.\+h.



Referenced by Oomph\+Lib\+Exception(), and $\sim$\+Oomph\+Lib\+Exception().

\mbox{\Hypertarget{classoomph_1_1OomphLibException_ac374cd3358bf26639860a6778779786d}\label{classoomph_1_1OomphLibException_ac374cd3358bf26639860a6778779786d}} 
\index{oomph::OomphLibException@{oomph::OomphLibException}!Suppress\_error\_message@{Suppress\_error\_message}}
\index{Suppress\_error\_message@{Suppress\_error\_message}!oomph::OomphLibException@{oomph::OomphLibException}}
\doxysubsubsection{\texorpdfstring{Suppress\_error\_message}{Suppress\_error\_message}}
{\footnotesize\ttfamily bool oomph\+::\+Oomph\+Lib\+Exception\+::\+Suppress\+\_\+error\+\_\+message\hspace{0.3cm}{\ttfamily [protected]}}



Boolean to suppress issuing of the error message in destructor (useful if error is caught successfully!) 



Definition at line 213 of file oomph\+\_\+definitions.\+h.



Referenced by disable\+\_\+error\+\_\+message(), Oomph\+Lib\+Exception(), and $\sim$\+Oomph\+Lib\+Exception().



The documentation for this class was generated from the following files\+:\begin{DoxyCompactItemize}
\item 
\mbox{\hyperlink{oomph__definitions_8h}{oomph\+\_\+definitions.\+h}}\item 
\mbox{\hyperlink{oomph__definitions_8cc}{oomph\+\_\+definitions.\+cc}}\end{DoxyCompactItemize}
